tests/test_config.py

```diff
@@ -753,3 +753,42 @@ def test_specifiers(tmp_path: Path) -> None:
 
 def test_deterministic() -> None:
     assert MkosiConfig.default() == MkosiConfig.default()
+
+
+def test_environment(tmp_path: Path) -> None:
+    d = tmp_path
+
+    (d / "mkosi.conf").write_text(
+        """\
+        [Content]
+        Environment=TestValue2=300
+                    TestValue3=400
+        EnvironmentFiles=other.env
+        """
+    )
+
+    (d / "mkosi.env").write_text(
+        """\
+        TestValue1=90
+        """
+    )
+
+    (d / "other.env").write_text(
+        """\
+        TestValue1=100
+        TestValue2=200
+        """
+    )
+
+    with chdir(d):
+        _, [config] = parse_config()
+
+        expected = {
+            "TestValue1": "100",
+            "TestValue2": "300",
+            "TestValue3": "400"
+        }
+
+        assert {k: v for k, v in config.environment.items() if k in expected} == expected
```

Comment:
Why the `if k in expected`? All keys from the env are in expected, so the only thing this can do is hide something unexpected, which a test should catch
```suggestion
        assert {k: v for k, v in config.environment.items()} == expected
```

Reply:
we have a couple more entries in the environment which is injected my mkosi, not in the the expected dict

Reply:
Good point! Though I'd express this with a slightly different focus and add a comment for why this looks the way it does
```suggestion
        # Only check values for keys from expected, as config.environment contains other items as well
        assert {k: config.environment[k] for k in expected.keys()} == expected
```

Reply:
aye aye